### ISSUE TITLE
get_schema_view parameters lack SchemaGenerator parameters

### ISSUE THREAD

**Issue by khornberg** [encode/django-rest-framework#4751]
## Checklist

- [ x ] I have verified that that issue exists against the `master` branch of Django REST framework.
- [ x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ x ] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [ x ] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [ x ] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce
Call `get_schema_view` with `patterns=something` and you get an error because it is not defined.

```
get_schema_view(title=None, url=None, renderer_classes=None):
     generator = SchemaGenerator(title=title, url=url)
 ```
[link](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/schemas.py#L574)

but `SchemaGenerator` object allows:

```
class SchemaGenerator(object):
    def __init__(self, title=None, url=None, patterns=None, urlconf=None)
```

## Expected behavior
Be able to call `get_schema_view` with `patterns`.
I don't know why it is not allowed given that the `SchemaGenerator` object that is instantiated allows `patterns` to limit the inspected urls. 

## Actual behavior
`TypeError` unexpected keyword argument

**Comment by m1kola** [comment]
The `SchemaGenerator` class is a public api, so you can call it directly.

Here is a shortcut I use to exclude unneeded endpoints:

```python
from django.conf.urls import include, url
from rest_framework.renderers import DocumentationRenderer, CoreJSONRenderer, SchemaJSRenderer

from rest_framework.schemas import SchemaGenerator, SchemaView


def include_docs_urls(title=None, description=None, patterns=None):
    generator = SchemaGenerator(title=title, description=description, patterns=patterns)

    docs_view = SchemaView.as_view(
        renderer_classes=[DocumentationRenderer, CoreJSONRenderer],
        schema_generator=generator,
        public=True,
    )
    schema_js_view = SchemaView.as_view(
        renderer_classes=[SchemaJSRenderer],
        schema_generator=generator,
        public=True,
    )

    urls = [
        url(r'^$', docs_view, name='docs-index'),
        url(r'^schema.js$', schema_js_view, name='schema-js')
    ]
    return include(urls, namespace='api-docs')
```

**Comment by xordoquy** [comment]
Will be solved by #4979
